lib/Log/Any/Adapter/DERIV.pm

```diff
@@ -619,6 +623,28 @@ sub level {
     return $num_to_name{$self->{log_level}};
 }
 
+=head2 set_context
+
+Set the log context - hash
+
+=cut
+
+sub set_context {
+    my ($self, $context) = @_;
+    $self->{context} = $context;
+}
+
+=head2 clear_context
+
+remove the log context - hash
+
+=cut
+
+sub clear_context {
```

Comment:
will it be better to restore context after `set_context` ?

Reply:
we set_context before logging a message and clear it afterwords, the reason behind this it to making sure used resources are cleared up and the most important being context is always correct as its passed by parameter for the request to Log